agenta-cli/agenta/sdk/agenta_decorator.py

```diff
@@ -90,7 +92,16 @@ async def wrapper_deployed(*args, **kwargs) -> Any:
             agenta.config.pull(config_name=kwargs["config"])
         else:
             agenta.config.pull(config_name="default")
-        return await execute_function(func, *args, **func_params)
+
+        config = agenta.config.all()
+        return await execute_function(
+            func,
+            *args,
+            **{
+                "params": func_params,
+                "config_params": {**config, "environment": kwargs["environment"]},
```

Comment:
> I would not put the environment in the config_params, since a user could environment as a name of a configuration.

I'm not sure I understand this.

> Also, environment is only used in the endpoint generate_deployed and not in generate
That is correct. When a user publishes their LLM app to an environment, we want to be able to trace the llm requests in the environment which it was published. The endpoint `generate` is only used in the playground, so when a user makes a call to the LLM app, the environment is set to `playground`.

Reply:
For the first part, I am saying that we should instead have
```
            **{
                "params": func_params,
                "config_params": **config,
                "environment": kwargs["environment"]

```
Because the user could already have the key environment in the config dict. Basically they could have done
```
ag.config.register_default(environment = ag.TextParam("some variable used in their prompt"))
```